src/main/java/listeners/MainEventListener.java

```diff
@@ -0,0 +1,232 @@
+
+    /**
+     * Check if the given Topic exists in the Server. Notifies the user if the topic does not exist.
+     * @param member The Member to reply to
+     * @param channel The Channel to reply within
+     * @param server The Server to check in
+     * @param topic The Topic to check for
+     * @return Contains the Topic, if it exists
+     */
+    private static Optional<Topic> checkTopicExists(Member member, TextChannel channel, Server server, String topic) {
+        Optional<Topic> topicOptional = Arrays.stream(server.getTopics())
+                .filter(t -> t.getName().equalsIgnoreCase(topic))
+                .findFirst();
+        if (topicOptional.isEmpty()) {
+            channel.sendMessage(String.format(
+                    "%s Topic \"%s\" does not exist.",
+                    member.getAsMention(),
+                    topic)).queue();
+        }
+        return topicOptional;
+    }
+
+    @Override
+    public void onReady(@NotNull ReadyEvent event) {
+        System.out.println("Logged in");
+        event.getJDA().getPresence().setPresence(Activity.playing("$help"), false);
+    }
+
+    @Override
+    public void onGuildMessageReceived(@NotNull GuildMessageReceivedEvent event) {
+        String[] tokens = event.getMessage().getContentDisplay().split(" ");
+        if (!tokens[0].startsWith("$")) return;
+
+        String guildID = event.getGuild().getId();
+        Server server = servers.computeIfAbsent(guildID, k -> Server.load(event.getGuild()));
+
+        // pick the correct method to call
+        CommandHandler commandHandler;
+        switch (tokens[0].substring(1)) {
+            case "help" -> commandHandler = this::help;
+            case "maketopic" -> commandHandler = this::maketopic;
+            case "deletetopic" -> commandHandler = this::deletetopic;
+            case "showtopics" -> commandHandler = this::showtopics;
+            case "queue" -> commandHandler = this::queue;
+            case "ready" -> commandHandler = this::ready;
+            case "showqueue" -> commandHandler = this::showqueue;
+            case "clear" -> commandHandler = this::clear;
+            default -> commandHandler = this::unknownCommand;
+        }
+        commandHandler.handle(
+                Objects.requireNonNull(event.getMember()),
+                event.getChannel(),
+                server,
+                Arrays.copyOfRange(tokens, 1, tokens.length));
+    }
+
+    private void help(Member member, TextChannel channel, Server server, String[] args) {
+        EmbedBuilder embedBuilder = new EmbedBuilder();
+        embedBuilder.setTitle("Help!");
+        embedBuilder.setDescription("Possible commands:");
+        embedBuilder.setColor(0xE57D25);
+
+        embedBuilder.addField("$queue <topic>", "Add yourself to a queue.", false);
+        embedBuilder.addField("$showqueue <topic>", "Show the people currently in queue.", false);
+        embedBuilder.addField("$showtopics", "List all topics.", false);
```

Comment:
Also what is the last bool parameter here?

Reply:
The last bool parameter determines whether the field is shown inline or not. In [this example](https://preview.redd.it/e7v6dvh5k3n41.png?width=607&format=png&auto=webp&s=37af4696cf03c623dc41684ec10f254511d09ee0), the first two fields are inline (`true`), while the second two are not inline (`false`).

Reply:
Ah kk!
I wish every language had named parameters...